v2/bigtable-cdc-to-hbase/src/main/java/com/google/cloud/teleport/v2/templates/transforms/ConvertChangeStream.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.transforms;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.cloud.bigtable.data.v2.models.ChangeStreamMutation;
+import com.google.cloud.bigtable.data.v2.models.DeleteCells;
+import com.google.cloud.bigtable.data.v2.models.DeleteFamily;
+import com.google.cloud.bigtable.data.v2.models.Entry;
+import com.google.cloud.bigtable.data.v2.models.Range.TimestampRange;
+import com.google.cloud.bigtable.data.v2.models.SetCell;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.beam.sdk.metrics.Metrics;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellScanner;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.RowMutations;
+import org.apache.hadoop.util.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Converts Bigtable change stream RowMutations objects to their approximating HBase mutations. */
+public class ConvertChangeStream {
+  private static final Logger LOG = LoggerFactory.getLogger(ConvertChangeStream.class);
+
+  /** Creates change stream converter transformer. */
+  public static ConvertChangeStreamMutation convertChangeStreamMutation() {
+    return new ConvertChangeStreamMutation();
+  }
+
+  /**
+   * Change stream converter that converts change stream mutations into RowMutation objects. Input
+   * is Bigtable change stream KV.of(rowkey, changeStreamMutation), Output is KV.of(rowkey, hbase
+   * RowMutations object)
+   */
+  public static class ConvertChangeStreamMutation
```

Comment:
For my education: do we need a PTransform? Can't we directly use a DoFn?

Reply:
We can - here we wrap it so it's encapsulated in case it needs to evolve in the future:
```
Expose every major data-parallel task accomplished by your library as a composite PTransform. This allows the structure of the transform to evolve transparently to the code that uses it: e.g. something that started as a ParDo can become a more complex transform over time.
```
- https://beam.apache.org/contribute/ptransform-style-guide/#exposing-a-ptransform-vs-something-else